blockchain/chainio.go

```diff
@@ -1587,12 +1587,6 @@ func (b *BlockChain) HeaderByHeight(height int64) (*wire.BlockHeader, error) {
 // block for the provided hash, deserialize it, retrieve the appropriate height
 // from the index, and return a dcrutil.Block with the height set.
 func dbFetchBlockByHash(dbTx database.Tx, hash *chainhash.Hash) (*dcrutil.Block, error) {
-	// First find the height associated with the provided hash in the index.
-	blockHeight, err := dbFetchHeightByHash(dbTx, hash)
```

Comment:
This can't be removed.  The height<->hash indices are how the code knows if the block is in the main chain or not.  Removing this means that check will be skipped, so it would lead to all types of incorrect results since this behavior is relied on.

So, given that, you could either leave the call and just ignore the height, or call `dbMainChainHasBlock` and produce the same error that calling `dbFetchHeightByHash` when it doesn't.